api/api.go

```diff
@@ -78,21 +78,33 @@ func (w Wrapper) AuthenticateWithPassword(ctx echo.Context) error {
 	if err != nil {
 		return ctx.JSON(http.StatusForbidden, errorResponse{err})
 	}
-	token, err := w.APIAuth.CreateSessionJWT(req.CustomerID, sessionId)
+	token, err := w.APIAuth.CreateSessionJWT(req.CustomerID, sessionId, false)
 	if err != nil {
 		return echo.NewHTTPError(http.StatusInternalServerError, err)
 	}
 
 	return ctx.JSON(200, domain.SessionToken{Token: string(token)})
 }
 
-func (w Wrapper) AuthenticateWithIRMA(ctx echo.Context) error {
-	req := domain.IRMAAuthenticationRequest{}
-	if err := ctx.Bind(&req); err != nil {
-		return ctx.JSON(http.StatusBadRequest, errorResponse{err})
+func (w Wrapper) getCustomerIDFromHeader(ctx echo.Context) (int, error) {
+	token, err := w.APIAuth.extractJWTFromHeader(ctx)
+	if err != nil {
+		ctx.Echo().Logger.Error(err)
+		return 0, echo.NewHTTPError(http.StatusUnauthorized, err)
+	}
+	rawID, ok := token.Get(CustomerID)
+	if !ok {
+		return 0, echo.NewHTTPError(http.StatusUnauthorized, "missing customerID in token")
 	}
+	return int(rawID.(float64)), nil
+}
```

Comment:
Didn't check it but we probably have some sort of auth middleware right? If we already validate the JWT we can store the CustomerID from the JWT in the echo context and get it from there instead.

Reply:
The auth middleware is only applied to the `/web/private` prefixed paths. These endpoinds listen on the `/web/auth` prefixes.

Reply:
I moved the extractJWTFromHeader to the api.Auth struct to declutter the api.go.